Patch-Source: https://www.postgresql.org/message-id/20220304.113347.2105652521035137491.horikyota.ntt@gmail.com (backported)
--
From: Kyotaro Horiguchi <horikyota.ntt@gmail.com>
Date: Fri, 4 Mar 2022 11:03:12 +0900
Subject: [PATCH] Wait for process termination during isolation tests

temp-schema-cleanup.spec used pg_terminate_backend() without specifying
timeout, thus there may be a case of proceeding to the next step before
the process actually terminates then false failure.

Supply timeout to pg_terminate_backend() so that it waits for process
termination to avoid that failure mode.

diff --git a/src/test/isolation/specs/temp-schema-cleanup.spec b/src/test/isolation/specs/temp-schema-cleanup.spec
index a9417b7e90..f0d3928996 100644
--- a/src/test/isolation/specs/temp-schema-cleanup.spec
+++ b/src/test/isolation/specs/temp-schema-cleanup.spec
@@ -47,7 +47,7 @@ step s1_discard_temp {
 }
 
 step s1_exit {
-    SELECT pg_terminate_backend(pg_backend_pid());
+    SELECT pg_terminate_backend(pg_backend_pid(), 180000);
 }
 
 
-- 
2.27.0
